pkg/apis/networking/types.go

```diff
@@ -56,13 +58,40 @@ type GroupMemberPod struct {
 	Ports []NamedPort
 }
 
+// ExternalEntityReference represents a ExternalEntity Reference.
+type ExternalEntityReference struct {
+	// The name of this ExternalEntity.
+	Name string
+	// The namespace of this ExternalEntity.
+	Namespace string
+}
+
+// ExternalEndpoint represents an external endpoint.
+type ExternalEndpoint struct {
```

Comment:
yes. In Pod, we have at most one IP and multiple ports; 
with ExternalEntity, we expands it include multiple ExternalEndpoints (see https://github.com/vmware-tanzu/antrea/pull/775/files#diff-838fb652afc3a48667ba93d07d803680),  each of ExternalEndpoint may include one IP and multiple ports.

Reply:
So, I might have misunderstood before. I thought ExternalEntity is a single interface, but do you mean mean it is a single VM? Then how we define per interface policies?
Or maybe you mean even a single interface can have multiple IPs?

Reply:
1. ExternalEntity may be derived from VMs, NativeServices, NetworkInterface, Ingress, Service is type = LoadBalance
2. Each of above resources generates a corresponding ExternalEntity.
3. For instance, if a VM contains 2 NetworkInterface, three ExternalEntities, one for VM, and one each for NetworkInterface, will be generated
4. Each of the ExternalEntities may potentially has multiple IPs, but in case of VM or NetworkInterface, it typically has one IP in the cloud
5. AntreaNetwork policies may be applied to VM, NetworkInterface, ...; depending on corresponding ExternalEntities selected in the AppliedToGroup.

Reply:
Ok. I now recall this. This sounds a little indirect and maybe not very easy to understand (both VM and VM interface are ExternalEntities). But I do not have another better idea to recommend, so I would not suggest any change (until I really have an idea).

Reply:
we can potentially using "kubectl get -o wide " with enough information so that user knows associated (cloud) resources .

another option discussed to create an upper tier cloud specific policy.

we will play with it once basic stuff is done, and see which option makes more sense.